guildwars2/characters.py

```diff
@@ -388,6 +401,33 @@ def handle_duplicates(upgrades):
         if character == "All":
             await user.send("\n".join(output))
 
+    @character.command(name="crafting")
+    @commands.cooldown(1, 10, BucketType.user)
+    async def character_crafting(self, ctx):
+        """Displays your characters and their crafting level"""
+        endpoint = "characters?page=0"
+        await ctx.trigger_typing()
+        try:
+            characters = await self.call_api(endpoint, ctx.author,
+                                             ["characters"])
+        except APIError as e:
+            return await self.error_handler(ctx, e)
+        data = discord.Embed(
+            description='Crafting overview', colour=self.embed_color)
+        for character in characters:
+            craftlist = ""
+            if character["crafting"] != []:
+                for crafting in character["crafting"]:
+                    rating = crafting["rating"]
+                    discipline = crafting["discipline"]
+                    craftlist += "\n".join(
+                        ["Level {0} {1}\n".format(rating, discipline)])
+                data.add_field(name=character["name"], value=craftlist)
```

Comment:
This doesn't do anything, you're using str.join() on a single element list, so nothing happens. This block can be replaced by this:
```py
for character in characters:
    craft_list = []
    if character["crafting"]:
        for crafting in character["crafting"]:
            rating = crafting["rating"]
            discipline = crafting["discipline"]
            craft_list.append("Level {} {}".format(rating, discipline))
            data.add_field(name=character["name"], value="\n".join(craft_list))
```
Same goes for the block above in character_info. Could even make it a function if you want.

Reply:
Created the function get_crafting() and used for both